src/ast/mod.rs

```diff
@@ -624,6 +590,12 @@ pub enum Expr {
     Identifier(Ident),
     /// Multi-part identifier, e.g. `table_alias.column` or `schema.table.col`
     CompoundIdentifier(Vec<Ident>),
+    /// Multi-part Expression accessing. It's used to represent an access chain from a root expression.
+    /// e.g. `expr[0]`, `expr[0][0]`, or `expr.field1.filed2[1].field3`, ...
```

Comment:
I think it would be awesome to update this comment with the information from this PR's description:

![Screenshot 2024-12-06 at 4 16 31 PM](https://github.com/user-attachments/assets/566860fb-fcb7-40ef-846b-70ccdffe0126)